File: app/schemas/calendar.py
from datetime import date
from uuid import UUID

from pydantic import BaseModel


class CalendarEvent(BaseModel):
    id: str
    title: str
    date: date
    category: str
    project_id: UUID | None = None
    milestone_id: UUID | None = None
    user_id: UUID | None = None


class ProjectTimelineBar(BaseModel):
    project_id: UUID
    tool_number: str
    customer_name: str | None = None
    team_id: UUID | None = None
    team_name: str | None = None
    health: str
    execution_status: str
    designer_name: str | None = None
    surfacer_name: str | None = None
    start_date: date
    end_date: date
    progress_percent: float = 0
    due_date_missing: bool = False

File: frontend/src/api/calendar.ts
import { apiClient } from './client';

export interface CalendarEvent {
  id: string;
  title: string;
  date: string;
  category: string;
  project_id?: string | null;
  milestone_id?: string | null;
  user_id?: string | null;
}

export interface ProjectTimelineBar {
  project_id: string;
  tool_number: string;
  customer_name?: string | null;
  team_id?: string | null;
  team_name?: string | null;
  health: string;
  execution_status: string;
  designer_name?: string | null;
  surfacer_name?: string | null;
  start_date: string;
  end_date: string;
  progress_percent: number;
  due_date_missing: boolean;
}

export async function fetchEngineeringCalendar(): Promise<CalendarEvent[]> {
  const { data } = await apiClient.get<CalendarEvent[]>('/calendar/engineering');
  return data;
}

export async function fetchProjectTimeline(params?: {
  from?: string;
  to?: string;
  team_id?: string;
}): Promise<ProjectTimelineBar[]> {
  const { data } = await apiClient.get<ProjectTimelineBar[]>('/calendar/project-timeline', {
    params,
  });
  return data;
}
